### PR TITLE
[doc]Job manager

### PR DESCRIPTION
**Pull request by Amber1990Zhang**
as title.
Code pr #1424

### CODE REVIEW

File: docs/manual-CN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
+# 作业管理（Job Manager）
+
+作业特指在存储层运行的两个命令 compact 和 flush。管理指对作业进行管理。比如让作业排队执行、查看作业状态、停止作业、恢复作业等。
```
Comment (wilsonyou) [inline docs/manual-CN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md]: `compact` `flush`

File: docs/manual-CN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+- `finished` 表示运行已结束，且成功。其他可能的状态有 Queue、running、failed、stopped
+- `12/17/19 17:21:30` 表示开始时间，初始为空(Queue)，当且仅当状态变为 running 时，才会设这个值
+- `12/17/19 17:21:30` 表示结束时间，如果为 Queue 或者 running 状态，这里会为空，当状态变为 finished、failed、stopped 时会设置此值
+- `40-0` 表示当前作业 ID 是 40，任务 ID 是 0
+- `192.168.8.5` 表示运行在 192.168.8.5 这台机器上
+- `finished` 表示运行已结束，且成功。其他可能的状态有 Queue、running、failed、stopped
+- `12/17/19 17:21:30` 表示开始时间，因为任务初始即为 running 状态，所以这里永不为空
+- `12/17/19 17:21:30` 表示结束时间，如果为 running 状态，这里会为空，当状态变为 finished、failed、stopped 时会设置此值
+
+**注意：** 作业状态有 QUEUE、RUNNING、FINISHED、FAILED、STOPPED 5 种。状态机转换（backuped 状态不会显示）见以下说明：
```
Comment (wilsonyou) [inline docs/manual-CN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md]: **注意：** 作业状态有五种，分别是 QUEUE、RUNNING、FINISHED、FAILED、STOPPED。状态机转换（backuped 状态不会显示）见以下说明：

File: docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
+# Job Manager
+
+The job here refers to the two commands `compact` and `flush` running at the storage layer. The manager means to manage the jobs. For example, you can run, show, stop and recover jobs.
```
Comment (wilsonyou) [inline docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md]: refers to two

File: docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+## Statements List
+
+### admin compact / flush
+
+`admin compact/flush` statement creates a new job and returns the job ID in the job manager, and executes the `compact/flush` command in the storage. The example is as follows:
```
Comment (wilsonyou) [inline docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md]: The `admin compact/flush`

File: docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+What's in the returned results:
+
+- `40` is the job ID
+- `flush nba` indicates that a flush operation is performed on space nba
+- `finished` is the job status, which indicates that the job execution is finished and successful. Other job statuses are Queue, running, failed and stopped
+- `12/17/19 17:21:30` is the start time, which is initially empty(Queue). The value is set if and only if the job status is running.
+- `12/17/19 17:21:30` is the stop time, which is empty when the job status is Queue or running. The value is set when the job status is finished, failed and stopped
+- `40-0` indicated that the job ID is 40, the task ID is 0
+- `192.168.8.5` shows which machine the job is running on
+- `finished` is the job status, which indicates that the job execution is finished and successful. Other job statuses are Queue, running, failed and stopped
```
Comment (wilsonyou) [inline docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md]: status is a more common practice

Please check out.

File: docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+Queue -- running -- finished -- backuped
+     \          \                /
+      \          \ -- failed -- /
+       \          \            /
+        \ --------- - stopped -/
+```
+
+#### List All Jobs
+
+The `SHOW JOBS` statement lists all the jobs (all the jobs that are not backuped, please refer to the next section on how to backup job).
```
Comment (wilsonyou) [inline docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md]: . Please refer to

File: docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+-----------------------------------------------------------------------------
+| 23     | compact test2 | stopped  | 12/06/19 15:07:09 | 12/06/19 15:07:33 |
+-----------------------------------------------------------------------------
+| 24     | compact test2 | stopped  | 12/06/19 15:07:11 | 12/06/19 15:07:20 |
+-----------------------------------------------------------------------------
+| 25     | compact test2 | stopped  | 12/06/19 15:07:13 | 12/06/19 15:07:24 |
+-----------------------------------------------------------------------------
+```
+
+For explanations on the returned results, please refer to the previous section [List Single Job Information](#list-single-job-information).
```
Comment (wilsonyou) [inline docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md]: For details about the returned results, refer to ...

File: docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+
+### BACKUP JOB
+
+The `BACKUP JOB <from_id> <to_id>`  statement archives a job when there are too many jobs. The archived jobs are no longer visible in the console. If you show an archived job, an error is thrown.
```
Comment (wilsonyou) [inline docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md]: archives jobs when there are

File: docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+| 24     | compact test2 | stopped  | 12/06/19 15:07:11 | 12/06/19 15:07:20 |
+-----------------------------------------------------------------------------
+| 25     | compact test2 | stopped  | 12/06/19 15:07:13 | 12/06/19 15:07:24 |
+-----------------------------------------------------------------------------
+```
+
+For explanations on the returned results, please refer to the previous section [List Single Job Information](#list-single-job-information).
+
+### BACKUP JOB
+
```
Comment (wilsonyou) [inline docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md]: If you use the Show command to display an archived job,

File: docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+=========================
+| BACKUP Result         |
+=========================
+| backup 1 jobs 2 tasks |
+-------------------------
+```
+
+### RECOVER JOB
+
+The `RECOVER JOB` statement re-execute the jobs and returns the number of the recovered jobs.
```
Comment (wilsonyou) [inline docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md]: re-executes

File: docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md (hunk 3 above)
Comment (Amber1990Zhang) [reply docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md]: i think it's ok here

**Comment by whitewum** [comment]
@liuyu85cn

File: docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+- `40` is the job ID
+- `flush nba` indicates that a flush operation is performed on space nba
+- `finished` is the job status, which indicates that the job execution is finished and successful. Other job status are Queue, running, failed and stopped
+- `12/17/19 17:21:30` is the start time, which is initially empty(Queue). The value is set if and only if the job status is running.
+- `12/17/19 17:21:30` is the stop time, which is empty when the job status is Queue or running. The value is set when the job status is finished, failed and stopped
+- `40-0` indicated that the job ID is 40, the task ID is 0
+- `192.168.8.5` shows which machine the job is running on
+- `finished` is the job status, which indicates that the job execution is finished and successful. Other job status are Queue, running, failed and stopped
+- `12/17/19 17:21:30` is the start time, which can never be empty because the initial status is running
+- `12/17/19 17:21:30` is the stop time, which is empty when the job status is running. The value is set when the job status is finished, failed and stopped
```
Comment (wilsonyou) [inline docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md]: end time

File: docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+Queue -- running -- finished -- removed
+     \          \                /
+      \          \ -- failed -- /
+       \          \            /
+        \ ---------- stopped -/
+```
+
+#### List All Jobs
+
+The `SHOW JOBS` statement lists all the jobs that are not expired. The default  job expiration time is one week. You can change it with meta flag `job_expired_secs`.
```
Comment (wilsonyou) [inline docs/manual-EN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md]: default job

File: docs/manual-CN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md [hunk 13]
```diff
@@ -0,0 +1,106 @@
+# 作业管理（Job Manager）
+
+作业特指在存储层运行的两个命令 `compact` 和 `flush`。管理指对作业进行管理。比如让作业排队执行、查看作业状态、停止作业、恢复作业等。
+
+## 命令列表
+
+### submit compact / flush
```
Comment (liuyu85cn) [inline docs/manual-CN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md]: submit job compact /  flush

File: docs/manual-CN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md [hunk 14]
```diff
@@ -0,0 +1,106 @@
+# 作业管理（Job Manager）
+
+作业特指在存储层运行的两个命令 `compact` 和 `flush`。管理指对作业进行管理。比如让作业排队执行、查看作业状态、停止作业、恢复作业等。
```
Comment (liuyu85cn) [inline docs/manual-CN/3.build-develop-and-administration/3.deploy-and-administrations/server-administration/storage-service-administration/job-manager.md]: Job 现在是定义为跑在storaged上的一些长任务。比如说compact 、 flush 等（rebuild index）

**Pull request merged**